pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -0,0 +1,108 @@
+package v1alpha1
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+
+// ServiceBindingSpec defines the desired state of ServiceBindingSpec
+type ServiceBindingSpec struct {
+	// MountPathPrefix is the prefix for volume mount
+	// +optional
+	MountPathPrefix string `json:"mountPathPrefix,omitempty"`
+
+	// EnvVarPrefix is the prefix for environment variables
+	// +optional
+	EnvVarPrefix string `json:"envVarPrefix,omitempty"`
+
+	// DataMapping is used to map CR/CSV/CRD attributes to Custom env variables
+	// +optional
+	DataMapping []corev1.EnvVar `json:"dataMapping,omitempty"`
+
+	// Services is used to identify multiple backing services.
+	Services []Service `json:"services"`
+
+	// Application is used to identify the application connecting to the
+	// backing service operator. When not provided, a managed binding secret
+	// is created but not bound to any application workload.
+	// +optional
+	Application Application `json:"application,omitempty"`
+
+	// DetectBindingResources is flag used to bind all non-bindable variables from
+	// different subresources owned by backing operator CR.
+	// +optional
+	DetectBindingResources bool `json:"detectBindingResources,omitempty"`
+}
+
+// ServiceBindingStatus defines the observed state of ServiceBinding
+// +k8s:openapi-gen=true
+type ServiceBindingStatus struct {
+	// Conditions describes the state of the operator's reconciliation functionality.
+	Conditions []conditionsv1.Condition `json:"conditions"`
+	// SecretRef is the name of the intermediate binding secret generated by
+	// the controller.
+	Secret corev1.LocalObjectReference `json:"secretRef"`
```

Comment:
Any particular reason as to the difference in the JSON tag `secretRef` and the `Secret` ?

Reply:
Nope, a miss from my side, I'll fix.

Reply:
Being fixed in 9b3f3d4